### PR TITLE
Cyber Implants Fall Update

### PR DESCRIPTION
**Pull request by xxalpha**
- Added user-implant interfacing (just feedback messaging for now); All /hud implants enable this functionality.
 - Added an /hud implant for interfacing, is EMP immune.
 - Buffed Reviver implant: faster healing, less cooldown.
 - Reduced the tech requirements/origins of Thermals and Reviver implants: Thermals now only need Syndicate 3.
 - Changed all cyber-implants material costs.
 - Fixed anti-stun implant emp_act() not calling parent.
 - Changed eye shield implant colors to be different than x-ray implant.

### CODE REVIEW

**Comment by Lumipharon** [comment]
origin tech to 3 means there's no viable way to get illegal 6 for adrenals anymore, without expensive traitor gear.

RIP, was hilarious while it lasted.

File: code/modules/research/designs/medical_designs.dm [hunk 1]
```diff
@@ -218,7 +229,7 @@
 	req_tech = list("materials" = 6, "programming" = 4, "biotech" = 7, "syndicate" = 4)
 	build_type = PROTOLATHE | MECHFAB
 	construction_time = 60
-	materials = list(MAT_METAL = 200, MAT_GLASS = 200, MAT_GOLD = 500, MAT_URANIUM = 1000, MAT_DIAMOND = 2000)
+	materials = list(MAT_METAL = 2000, MAT_GLASS = 2000, MAT_GOLD = 6000, MAT_URANIUM = 2000, MAT_DIAMOND = 4000)
```
Comment (Core0verload) [inline code/modules/research/designs/medical_designs.dm]: That's a HUGE price lift going on here.

Comment (xxalpha) [inline code/modules/research/designs/medical_designs.dm]: It's practically nothing with upgraded machines.

And even then, it's one of the best implants and it just got buffed too.

**Comment by Core0verload** [comment]
User-implant interfaces are pointless. Some of your code would conflict with my organs update 3.

**Comment by xxalpha** [comment]
No they're not.

**Comment by Core0verload** [comment]
Yes they are. They only add some feedback messages... wait no, they more like remove some feedback messages and remove implants action buttons from users without HUD. And current implants have NO action buttons.

I like the idea of complex implants having control panels, but your changes just suck.

**Comment by xxalpha** [comment]
So you want to magically enable cybernetic implants in your body? Humans are not cyborgs.

**Comment by Core0verload** [comment]
Why magically? Just connect them to neural system. It's already possible to create arm protesis with feedback, all controlled by mind, and we are in 2015. Imagine what is possible in 2500+.

**Comment by xxalpha** [comment]
And messages to the human are supposed to be what, voices in their head?

**Comment by Core0verload** [comment]
Any kind of body feedback. Pressure, heat, pain, internal muscular sensors - neural interface can emulate anything your body has and a little more. Did you know that humans can recieve visual signals from electrodes placed on *tounge*? Human neural system is insanely flexible.

**Comment by Shadowlight213** [comment]
Have subtle feedback with no interface, and more detailed feedback with it.
Also emp should cause the interface implant to display random ic ads, because that would be hilarious.

**Comment by Razharas** [comment]
@xxalpha

**Comment by xxalpha** [comment]
I have two more months to fix this :^)

**Comment by Razharas** [comment]
@xxalpha 
cmon

**Comment by optimumtact** [comment]
good changes here pls

**Comment by GunHog** [comment]
Wake up @xxalpha!

**Comment by Razharas** [comment]
Month old with no progress - closed

**Pull request closed without merging**